theHarvester/discovery/virustotal.py

```diff
@@ -64,21 +80,36 @@
     async def parse_hostnames(data, word):
         total_subdomains = set()
         for attribute in data:
-            total_subdomains.add(attribute['id'].replace('"', '').replace('www.', ''))
-            attributes = attribute['attributes']
+            total_subdomains.add(attribute["id"].replace('"', "").replace("www.", ""))
+            attributes = attribute["attributes"]
             total_subdomains.update(
-                {value['value'].replace('"', '').replace('www.', '') for value in attributes['last_dns_records'] if
-                 word in value['value']})
-            if 'last_https_certificate' in attributes.keys():
-                total_subdomains.update({value.replace('"', '').replace('www.', '') for value in
-                                         attributes['last_https_certificate']['extensions']['subject_alternative_name']
-                                         if word in value})
+                {
+                    value["value"].replace('"', "").replace("www.", "")
+                    for value in attributes["last_dns_records"]
+                    if word in value["value"]
+                }
+            )
+            if "last_https_certificate" in attributes.keys():
+                total_subdomains.update(
+                    {
+                        value.replace('"', "").replace("www.", "")
+                        for value in attributes["last_https_certificate"]["extensions"][
+                            "subject_alternative_name"
+                        ]
+                        if word in value
+                    }
+                )
         total_subdomains = list(sorted(total_subdomains))
         # Other false positives may occur over time and yes there are other ways to parse this, feel free to implement
         # them and submit a PR or raise an issue if you run into this filtering not being enough
         # TODO determine if parsing 'v=spf1 include:_spf-x.acme.com include:_spf-x.acme.com' is worth parsing
-        total_subdomains = [x for x in total_subdomains if not str(x).endswith('edgekey.net') and not str(x).endswith(
-            'akadns.net') and 'include:_spf' not in str(x)]
+        total_subdomains = [
+            x
+            for x in total_subdomains
+            if not str(x).endswith("edgekey.net")
+            and not str(x).endswith("akadns.net")
```

Comment:
```py
# str_parse.py

from timeit import timeit

cases = [
    '',
    'edgekey.net',
    'edgekey.net/path/at/end',
    'sub.domain.before.edgekey.net',
    'sub.domain.edgekey.net/and/path',
    'just some long string without the url in it'
]

for case in cases:
    print(timeit(lambda: not case.endswith('edgekey.net')))

print()

for case in cases:
    print(timeit(lambda: 'edgekey.net' not in case))
```
```
❯ python3 str_parse.py
0.10227041599999999
0.100186459
0.09366120799999997
0.10369545800000002
0.09712562500000005
0.09975475

0.04075262499999999
0.05183016600000001
0.05212399999999995
0.053757500000000014
0.062358625
0.04864037500000007
```
```py
# theHarvester.discovery.virustotal.SearchVirustotal.parse_hostnames

total_subdomains = [
    x
    for x in total_subdomains
    if "edgekey.net" not in str(x)
    and "edgesuite.net" not in str(x)
    and "include:_spf" not in str(x)
]
```
```
❯ pytest              
======================================= test session starts ========================================
platform darwin -- Python 3.9.6, pytest-7.4.0, pluggy-1.2.0
rootdir: /Users/yunis/repos/theHarvester
configfile: pytest.ini
testpaths: tests
plugins: asyncio-0.21.1, anyio-3.7.1, typeguard-2.13.3
asyncio: mode=auto
collected 16 items                                                                                 

tests/test_myparser.py .                                                                     [  6%]
tests/discovery/test_anubis.py ...                                                           [ 25%]
tests/discovery/test_certspotter.py ...                                                      [ 43%]
tests/discovery/test_githubcode.py .....                                                     [ 75%]
tests/discovery/test_otx.py ..                                                               [ 87%]
tests/discovery/test_threatminer.py ..                                                       [100%]

================================== 16 passed in 80.41s (0:01:20) ===================================
```
`parse_hostnames` was brought in on commit [Fixed VirusTotal module (#1172)](https://github.com/laramies/theHarvester/commit/c801db67255764fccac511208f6581c3079145cd#diff-ff6284b7a476aaaad5cf54ae850ec8750cf254b8a9a1c328c848850c7157a62dR65).
Making this change to is `in` rather than `str.endswith()` will explicitly change the functionality, but that's also what the CodeQL wants. 
It also comes with a performance boost, so no worry about slowing anything down. 
If this sounds good, I can push this change.

Reply:
Yeah push this change please, performance boost and fixing codeql cannot go wrong there